### ISSUE TITLE
having an executable script in a Pre Condition

### ISSUE THREAD

**Issue by ufrhaidar** [BehaviorTree/BehaviorTree.CPP#895]
Similar functionality to `_post`.

It is good for many things as it can help the tree be more readable:
- resetting or making sure a variable exists
- updating status to know a node has started.

**Comment by facontidavide** [comment]
yes, I have been thinking about that. I think that I will add it

**Comment by Aglargil** [comment]
> yes, I have been thinking about that. I think that I will add it

This PR might help solve this case. Let me know if you need any clarification on the implementation.
https://github.com/BehaviorTree/BehaviorTree.CPP/pull/899

**Comment by facontidavide** [comment]
yes, I also what this. will be added to version 5
